go_metrics/metrics/graphite.py

```diff
@@ -141,12 +158,68 @@ def get(self, **kw):
         null_parser = null_parsers[params['nulls']]
         returnValue(self._parse_response((yield resp.json()), null_parser))
 
+    @inlineCallbacks
+    def fire(self, **kw):
+        metrics = []
+        metrics_values = []
+        for mname, mvalue in kw.iteritems():
+            metric_name = self._get_full_metric_name(mname)
+            aggregator = self.aggregators.get(
+                agg_from_name(metric_name), AVG)
+
+            try:
+                mvalue = float(mvalue)
+            except (ValueError, TypeError):
+                raise BadMetricsQueryError(
+                    '%r is not a valid metric value,'
+                    'should be a floating point number' % mvalue)
+
+            metrics.append((Metric(metric_name, [aggregator]), mvalue))
+            metrics_values.append({
+                'name': mname,
+                'value': mvalue, 
+                'aggregator': aggregator.name,
+            })
+
+        mm = yield self.backend.worker.metric_manager
+        [mm.oneshot(metric, value) for metric, value in metrics]
+
+        returnValue(metrics_values)
+
+
+class MetricWorker(Worker):
+    def __init__(self, *args, **kwargs):
+        super(MetricWorker, self).__init__(*args, **kwargs)
+        self._started_d = Deferred()
+
+    @inlineCallbacks
+    def startWorker(self):
+        self._metric_manager = yield self.start_publisher(
+            MetricManager, '%s.' % self.config.get('prefix'))
+        self._started_d.callback(self._metric_manager)
+
+    @inlineCallbacks
+    def stopWorker(self):
+        if not self._started_d.called:
+            yield self._started_d
+        self._metric_manager.stop()
+
+    @property
```

Comment:
I sort of feel that since this returns a deferred it should be a method and not a property. Just a vague preference though.